sqlmesh/core/plan/definition.py

```diff
@@ -783,7 +783,9 @@ def is_finished(self) -> bool:
         return self == PlanStatus.FINISHED
 
 
-class SnapshotIntervals(PydanticModel, frozen=True):
+# millions of these can be created, pydantic has significant overhead
```

Comment:
Where do millions come from? Isn't it just one per snapshot?

Reply:
i gues intervals can be a lot and then pydantic validates it. changing to data class reduces plan by 5x in my tests